tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/findandreplace/TestFindReplaceLogic.java

```diff
@@ -0,0 +1,275 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.findandreplace;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.After;
+import org.junit.Test;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+
+import org.eclipse.jface.text.Document;
+import org.eclipse.jface.text.TextViewer;
+
+public class TestFindReplaceLogic {
+
+	Shell parentShell;
+	IFindReplaceLogic findReplaceLogic;
+	TextViewer textViewer;
```

Comment:
The fields should be initialized in some general setup functionality (called before each test execution) or should be moved to local variables in the test methods. Now it is unclear when one of these fields can expected to be not null, as you have to inspect the `setup...` methods to know which of them initializes them.